awx/main/dispatch/pool.py

```diff
@@ -74,7 +83,8 @@ def put(self, body):
             if not body.get('uuid'):
                 body['uuid'] = str(uuid4())
             uuid = body['uuid']
-        self.managed_tasks[uuid] = body
+        if self.track_managed_tasks:
+            self.managed_tasks[uuid] = body
```

Comment:
This should also cut down notably on memory consumption for high workloads that have a backlog cc @AlanCoding